---
title: "Ошибка компилятора CS0155 | Microsoft Docs"
ms.custom: ""
ms.date: "11/17/2016"
ms.prod: "visual-studio-dev14"
ms.reviewer: ""
ms.suite: ""
ms.technology: 
  - "devlang-csharp"
ms.tgt_pltfrm: ""
ms.topic: "article"
f1_keywords: 
  - "CS0155"
dev_langs: 
  - "CSharp"
helpviewer_keywords: 
  - "CS0155"
ms.assetid: 6c92984a-2b10-453e-9cb7-e6a1d1b98aa6
caps.latest.revision: 8
caps.handback.revision: 8
author: "BillWagner"
ms.author: "wiwagn"
manager: "wpickett"
translationtype: Human Translation
---
# Ошибка компилятора CS0155
Тип в операторах caught или thrown должен быть производным от System.Exception.  
  
 Выполнена попытка передачи типа данных, который не является производным от **System.Exception**, в блок [catch](../../csharp/language-reference/keywords/try-catch.md). В блок **catch** можно передавать только типы данных, которые являются производными от **System.Exception**. Дополнительные сведения см. в разделах [Операторы обработки исключений](../../csharp/language-reference/keywords/exception-handling-statements.md) и [Исключения и обработка исключений](../../csharp/programming-guide/exceptions/exceptions-and-exception-handling.md).  
  
 Следующий пример приводит к возникновению ошибки CS0155:  
  
```  
// CS0155.cs using System; namespace MyNamespace { public class MyClass2 // try the following line instead // public class MyClass2 : Exception { } public class MyClass { public static void Main() { try { } catch (MyClass2)   // CS0155, resolves if you derive MyClass2 from Exception { } } } }  
```